### PR TITLE
HIVE-21956 : Add the list of table selected by dump in the dump folder.

### PR DESCRIPTION
**Pull request by maheshk114**

### CODE REVIEW

File: ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplDumpTask.java [hunk 1]
```diff
@@ -376,13 +405,68 @@ private ReplicationSpec getNewEventOnlyReplicationSpec(Long eventId) {
...
+      tableListFile = new Path(dbRoot, ReplUtils.REPL_TABLE_LIST_FILE_NAME);
+      tableListFile = new Path(tableListFile, dbName.toLowerCase());
+    } else {
+      tableListFile = new Path(dbRoot, dbName.toLowerCase());
+      tableListFile = new Path(tableListFile, ReplUtils.REPL_TABLE_LIST_FILE_NAME);
+    }
+
+    int count = 0;
+    while (count < FileUtils.MAX_IO_ERROR_RETRY) {
+      try (FSDataOutputStream writer = FileSystem.get(hiveConf).create(tableListFile)) {
```
Comment (sankarh) [inline ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplDumpTask.java]: In retry mode, create might fail. Should we check if file exists and use open (overwrite mode) instead?

File: ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplDumpTask.java [hunk 2]
```diff
@@ -219,13 +224,33 @@ private boolean shouldBootstrapDumpTable(Table table) {
...
+  }
+
   private Long incrementalDump(Path dumpRoot, DumpMetaData dmd, Path cmRoot, Hive hiveDb) throws Exception {
     Long lastReplId;// get list of events matching dbPattern & tblPattern
     // go through each event, and dump out each event to a event-level dump dir inside dumproot
     String validTxnList = null;
     long waitUntilTime = 0;
     long bootDumpBeginReplId = -1;
 
+    List<String> tableList = work.replScope.includeAllTables() ? null : new ArrayList<>();
```
[the remark this answers is not in the document]
Comment (maheshk114) [reply ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplDumpTask.java]: yes ..if file is not present then they assume it to be db level replicattiion

File: ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplDumpTask.java [hunk 3]
```diff
@@ -376,13 +405,68 @@ private ReplicationSpec getNewEventOnlyReplicationSpec(Long eventId) {
     return rspec;
   }
 
-  Long bootStrapDump(Path dumpRoot, DumpMetaData dmd, Path cmRoot, Hive hiveDb) throws Exception {
+  private void dumpTableListToDumpLocation(List<String> tableList, Path dbRoot, String dbName, boolean isIncLoad,
+                                           HiveConf hiveConf) throws IOException, LoginException {
+    // Empty list will create an empty file to distinguish it from db level replication. If no file is there, that means
+    // db level replication. If empty file is there, means no table satisfies the policy.
+    if (tableList == null) {
```
[the remark this answers is not in the document]
Comment (maheshk114) [reply ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplDumpTask.java]: empty list will create a file with no table name

File: ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplDumpTask.java [hunk 4]
```diff
@@ -376,13 +405,68 @@ private ReplicationSpec getNewEventOnlyReplicationSpec(Long eventId) {
...
+      LOG.debug("Table list file is not created for db level replication.");
+      return;
+    }
+
+    // For incremental dump the table list is dumped in _tables/dbname file and for bootstrap dump, its dumped in
```
[the remark this answers is not in the document]
Comment (maheshk114) [reply ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplDumpTask.java]: then one directory has to be created for each db .. i think this looks fine and it was discussed in the stand up ..to be like this

File: ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplDumpTask.java (hunk 1 above)
Comment (maheshk114) [reply ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplDumpTask.java]: create is by default overwrite mode ..if file is there ..it over writes

File: ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplDumpTask.java [hunk 5]
```diff
@@ -219,13 +224,33 @@ private boolean shouldBootstrapDumpTable(Table table) {
     return !ReplUtils.tableIncludedInReplScope(work.oldReplScope, table.getTableName());
   }
 
+  private boolean isTableSatifiesConfig(Table table) {
```
[the remark this answers is not in the document]
Comment (maheshk114) [reply ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplDumpTask.java]: i think ..its a very simple function ..the intend can be understood by reading the code :)

**Pull request closed without merging**